src/plugins/blockbook.ts

```diff
@@ -85,14 +93,22 @@ export function makeBlockbook(opts: BlockbookOptions): AddressPlugin {
     })
     ws.on('close', () => {
       pluginDisconnectionCounter.inc({ pluginId, url: safeUrl })
-      codec.handleClose()
-      // Remove connection from connections array
-      connections.splice(connections.indexOf(connection), 1)
-      // Remove connection from addressToConnection map
-      for (const address of connection.addresses) {
-        addressToConnection.delete(address)
+
+      if (connection === blockConnection) {
+        // If this was the block connection, re-init it.
+        blockConnection = null
+        initBlockConnection()
```

Comment:
Do we want an exponential backoff here? We don't need the exponential backoff for the address connections because the remote app re-subscribing should create a nice delay. With the block connection, though, we could thrash a bit harder.

Maybe we could add a 1-second snooze call, which doubles to 2 seconds, 4, seconds, etc. until we hit 60 seconds or so max. If we get a successful `{ subscribed: true }` response to our block subscription request, then we reset the delay to 1, since we know the server is back online.

We don't have to do this today, so a simple `// TODO: Throttle reconnection attempts` comment would be fine.